python_transport/wirepas_gateway/protocol/mqtt_wrapper.py

```diff
@@ -145,22 +160,24 @@ def _get_socket(self):
             self.logger.error("Impossible to connect - authentication failure ?")
             return None
 
-        # Socket is not opened anymore, try to reconnect
-        timeout = MQTTWrapper.TIMEOUT_RECONNECT_S
-        while timeout > 0:
+        # Socket is not opened anymore, try to reconnect for timeout if set
+        timeout = self.timeout
+        no_timeout = (timeout == 0)
+
+        while no_timeout or timeout > 0:
```

Comment:
Am I wrong or could this comparison be simplified further?

`while timeout ==0 or timeout >0` => `while timeout =>0`

Reply:
No, it is maybe not well written but timeout and no_timeout is really different.

In fact timeout is initialized to self.timeout (being the max value set by user) and is updated in the loop.
But in case self.timeout is 0 (meaning reconnect forever), timeout is bypassed and it is a while True.

But as you raised the point, I will try to make it more clear to avoid confusion

Reply:
Thanks :cookie: